pkg/query-service/app/logparsingpipeline/pipelineBuilder.go

```diff
@@ -169,3 +214,151 @@ func cleanTraceParser(operator *PipelineOperator) {
 		operator.TraceFlags = nil
 	}
 }
+
+// Generates an expression checking that `fieldPath` has a non-nil value in a log record.
+func fieldNotNilCheck(fieldPath string) (string, error) {
+	_, err := expr.Compile(fieldPath)
+	if err != nil {
+		return "", fmt.Errorf("invalid fieldPath %s: %w", fieldPath, err)
+	}
+
+	// helper for turning `.` into `?.` in field paths.
+	// Eg: a.b?.c.d -> a?.b?.c?.d
+	optionalChainedPath := func(path string) string {
+		return strings.ReplaceAll(
+			strings.ReplaceAll(path, "?.", "."), ".", "?.",
+		)
+	}
+
+	// Optional chaining before membership ops is not supported by expr.
+	// Eg: The field `attributes.test["a.b"].value["c.d"].e` can't be checked using
+	// the nil check `attributes.test?.["a.b"]?.value?.["c.d"]?.e != nil`
+	// This needs to be worked around by checking that the target of membership op is not nil first.
+	// Eg: attributes.test != nil && attributes.test["a.b"]?.value != nil && attributes.test["a.b"].value["c.d"]?.e != nil
```

Comment:
I'm working on adding support for `foo?.["bar"]` in Expr right now. Will be done in the next version. ;)

Reply:
As well as on mode in which all `.` operations will be set to `?.`.

Reply:
Strange. I'm getting BinaryNode:
<img width="897" alt="Screenshot 2024-01-16 at 16 47 22" src="https://github.com/SigNoz/signoz/assets/141232/f64e11d6-2733-4b28-81cc-dd0a84543050">